src/app/tests/test-interaction-model-api.cpp

```diff
@@ -149,6 +157,12 @@ std::optional<ActionReturnStatus> TestImCustomDataModel::Invoke(const InvokeRequ
     return std::make_optional<ActionReturnStatus>(CHIP_ERROR_NOT_IMPLEMENTED);
 }
 
+ProviderChangeListener & TestImCustomDataModel::GetAttributeChangeReporter()
+{
+    static TestProviderChangeListener changeListener;
```

Comment:
why static? Since this is a test, could this thing be a property of the object itself? We could make it accumulate a std::vector of attributepathparams passed in, so we can validate their values.